Python/Product/PythonTools/PythonTools/Intellisense/ProjectAnalyzer.cs

```diff
@@ -236,7 +237,7 @@ IPythonInterpreterFactory factory
                 Trace.TraceInformation($"Connection log: {_conn.LogFilename}");
             }
 
-            Task.Run(() => _conn.ProcessMessages().HandleAllExceptions(_services.Site, allowUI: false)).DoNotWait();
+            _processingTask = Task.Run(() => _conn.ProcessMessages().WaitAndHandleAllExceptions(_services.Site, allowUI: false));
```

Comment:
Synchronously waiting for task inside a `Task.Run` isn't a good idea. It may end up in a deadlock even in SyncContext-free environment. 
`_processingTask = Task.Run(() => _conn.ProcessMessages()).HandleAllExceptions(_services.Site, allowUI: false)` should be enough.

But this doesn't answer the question what was the problem during test run. `TaskObserver` waits several seconds before it throws. Is it a problem only with default analyzer?

Reply:
Yes. The default analyzer stays running for the VS instance, and we reuse the VS instance between successful tests, so the analyzer is running still. So the problem is that our test infrastructure is waiting on a task that isn't meant to complete.

I _think_ you're right that we don't need to wait on this task, but let me check it before changing.